skaffold.md

```diff
@@ -0,0 +1,297 @@
+# Skaffold
+
+Skaffold is a command line utility that facilitates continuous development for Kubernetes applications. With Skaffold, you can iterate on your application source code locally then deploy it to a remote Kubernetes clusters.  Skaffold handles the workflow for building, pushing, and deploying your application.  It can also be used in an automated context such as a CI/CD pipeline to leverage the same workflow and tooling when moving applications to production.
+
+## Operating modes
+
+Skaffold has 2 operating modes: `skaffold dev` and `skaffold run`.
+
+### skaffold dev
+
+Updates your deployed application continually:
+
+* Watches your source code and the dependencies of your docker images for changes and runs a build and deploy when changes are detected
+* Streams the logs from deployed containers to your terminal
+* Creates a continuous build-deploy loop that only warn on errors
+
+### skaffold run
+
+Runs a Skaffold pipeline once, exits on any errors in the pipeline. Use for:
+
+* Continuous integration or continuous deployment pipelines
+* Sanity checking after iterating on your application
+
+## Pluggability
+
+Skaffold has a pluggable architecture that allows you to choose which tools you want to use.  In the following exercise, you will build the container locally using the Docker daemon, push it to an ECR registry, and deploy it to EKS by applying a Kubernetes manifest. 
+
+![plugabbility](https://github.com/jicowan/aws-microservices-deploy-options/blob/master/images/skaffold.jpg)
+
+## Prerequisites
+
+### Install ECR Credential Helper
+
+The ECR credential helper makes it easier to use ECR by seamless passing your AWS credentials to the service.  When you use the credential helper there is no need to use `docker login` or `docker logout`.  
+
+> If you're planning to use the credential helper with an assumed role, you'll need to set the environment variable `AWS_SDK_LOAD_CONFIG=true`.  The IAM principle you're using has to have permission to ECR too. 
+
+Install the credential helper with  `go get`:
+
+```
+`go get -u github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/docker-credential-ecr-login`
```

Comment:
Formatting is a bit weird here. No need for additional tick in the beginning of line 41.